src/main/java/uk/gov/justice/digital/job/DataHubJob.java

```diff
@@ -13,10 +13,7 @@
 import uk.gov.justice.digital.converter.Converter;
```

Comment:
writeInvalidRecords() method should be moved to Zone class as abstract method and implement the method in All zone class where Zone class is getting extended. Zone extending DeltaWriter is better

Reply:
I disagree Karthik. 
Zone is an interface. 
There is little that each Zone shares. Raw just appends; Structured handles validation and upsert/merges to storage. Curated currently looks like Structured but will functionally move away.
DeltaWriter should be changed to ZoneWriter, to avoid confusion.

Reply:
Java does not support inheriting multiple classes so making `Zone` an abstract class will not compile.

Not every zone writes invalid records so putting `writeInvalidRecords()` in `Zone` would require making `Zone` an abstract class as you suggested. Unfortunately, this will break the compilation as mentioned earlier.